_posts/2019-07-10-feature-columns/feature_columns.Rmd

```diff
@@ -0,0 +1,566 @@
+---
+title: "TensorFlow feature columns: Transforming your data recipes-style"
+description: > 
+ TensorFlow feature columns provide useful functionality for preprocessing categorical data and chaining transformations like bucketization or feature crossing. From R, we use them in popular "recipes" style, creating and subsequently refining a feature specification. In this post, we show how using feature specs frees cognitive resources and lets you focus on what you really want to accomplish. What's more, because of its elegance, feature-spec code reads nice and is fun to write as well. 
+author:
+  - name: Daniel Falbel and Sigrid Keydana
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: falbelkeydana2019featurecols
+date: 07-10-2019
+bibliography: bibliography.bib
+categories:
+  - Keras
+  - Introductions
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/feature_cols_hier.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+It's 2019; no-one doubts the effectiveness of deep learning in computer vision. Or natural language processing. With "normal", Excel-style, a.k.a. _tabular_ data however, the situation is different. 
+
+Basically there are two cases: One, you have numeric data only. Then, creating the network is straightforward, and all will be about optimization and hyperparameter search. Two, you have a mix of numeric and categorical data, where categorical could be anything from ordered-numeric to symbolic (e.g., text). In this latter case, with categorical data entering the picture, there is an extremely nice idea you can make use of: _embed_ what are equidistant symbols into a high-dimensional, numeric representation. In that new representation, we can define a distance metric that allows us to make statements like "cycling is closer to running than to baseball", or "`r emo::ji("smiley")` is closer to `r emo::ji("joy")` than to `r emo::ji("angry")`". When not dealing with language data, this technique is referred to as _entity embeddings_ ^[see e.g. [Entity embeddings for fun and profit](https://blogs.rstudio.com/tensorflow/posts/2018-11-26-embeddings-fun-and-profit/)].
```

Comment:
actually I'd like to skip this because these should come across differently-weighted sort of, you know -- kinda just mentioning case 1 but stressing case 2